cogs/commands/roomsearch.py

```diff
@@ -0,0 +1,99 @@
+from io import BytesIO
+
+import discord
+import requests
+from discord.ext import commands
+from discord.ext.commands import Bot, Context
+
+LONG_HELP_TEXT = """
+Finds a room on the various Warwick systems
+"""
+
+SHORT_HELP_TEXT = """Warwick Room Search"""
+
+
+class RoomSearch(commands.Cog):
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @commands.command(help=LONG_HELP_TEXT, brief=SHORT_HELP_TEXT)
+    async def room(self, ctx: Context, name: str):
+        rooms = self.get_room_infos(name)
+        if not rooms:
+            return await ctx.reply(
+                "Room does not exist. Ensure you are giving the full name"
+            )
+
+        if len(rooms) > 1:
+            room = await self.choose_room(ctx, rooms)
+            if room is None:
+                return
+        else:
+            room = rooms[0]  # Only one in rooms
+
+        desc = f"Building: **{room['building']}** {room['floor']}\n"
+        desc += f"**[Campus Map](https://campus.warwick.ac.uk/?cmsid={room['id']})**\n"
+        desc += f"**[Room Info (if centrally timetabled)](https://warwick.ac.uk/services/its/servicessupport/av/lecturerooms/roominformation/{room['name'].replace('.', '')})**\n"
+        desc += f"`Timetable coming soon?`\n"
+        desc += f"[Warwick Search](https://search.warwick.ac.uk/?q={name}) Room Capacity: {room['roomCapacity']}\n"
```

Comment:
This can be done using "\n".join(parts)

Reply:
Missing commas to delimit the items, which will be concatenated. Join also requires an iterable, i.e. a list or tuple here.